designs/carbon-aware.md

```diff
@@ -0,0 +1,133 @@
+# Carbon Aware Karpenter: Optimizing Kubernetes Cluster Autoscaling for Environmental Sustainability
+*Author: [@JacobValdemar](https://github.com/JacobValdemar)*
+
+## Context & Problem
+
+There is a growing concern about environmental sustainability within the context of Kubernetes cluster autoscaling. In multiple comments on [the proposal for moving Karpenter to CNCF](https://github.com/kubernetes/org/issues/4258), the move is backed because of opportunities within environmental sustainability.
+
+I'm currently working on my master's thesis in Computer Engineering (M.Sc.Eng) at Aarhus University located in Denmark. The objective of the thesis is to enable Karpenter to minimize carbon emissions from Kubernetes clusters that run on cloud infrastructure (focus is AWS).
+
+RFC: https://github.com/aws/karpenter/issues/4630
+
+## Fundamentals of Green Software
+I'll try to keep it simple. The reader should be familiar with the following.
+
+A cluster's emissions is made of two elements: embodied emissions and operational emissions. To get the total emissions, one can add them togeather.
+
+- **Embodied carbon emissions**: Manufacturing emissions (CO2e) amortized over instance lifetime (usually 4 years) divided by how long we use the instance
```

Comment:
Do any cloud providers publish this data? Without it, the optimisation is hard to do.

Reply:
Unfortunately, cloud providers generally do not publish high-quality carbon emission data. However, as noted in the _Carbon emissions data source_ section, there are still valid methods to obtain carbon emission data. An example is Boaviztapi which you can interact with in the [data demo](https://dataviz.boavizta.org/cloudimpact) and read about their methodology in [this article](https://www.boavizta.org/en/blog/empreinte-de-la-fabrication-d-un-serveur).